packages/client/hmi-client/src/App.vue

```diff
@@ -45,21 +45,28 @@ const dataExplorerActivated = ref(false);
  */
 const project = ref<Project | null>(null);
 
+async function setProject(id: Project['id']) {
+	// If the Id or the Project are null,
+	// set the Project to null, and send the user to the homepage.
+	if (!id) {
+		project.value = null;
+		goToHomepage();
+	} else {
+		project.value = await ProjectService.get(id);
+		if (!project.value) {
+			goToHomepage();
+		}
+	}
+}
+
+// Set the project when creating the component,
+// and anytime the URL is updated.
+setProject(route.params.projectId as string);
 watch(
 	() => route.params.projectId,
 	async (projectId) => {
-		if (!projectId) {
-			// Set the Project to null,
-			// and send the user to the homepage.
-			project.value = null;
-			goToHomepage();
-		} else {
-			const id = projectId as string;
-			project.value = await ProjectService.get(id);
-			if (!project.value) {
-				goToHomepage();
-			}
-		}
+		const id = projectId as string;
+		setProject(id);
 	}
```

Comment:
I was thinking something like below would be an easier fix, in the sense that if we need to track more variables like modelId or simulationId...this provides a cleaner way to read the logic, and potentially less chance s we'd clobber our watched variables together. 

Basically we just rig the original watch's third option to have `{immediate: true}`.


```
watch(
	() => route.params.projectId,
	async (projectId) => {
		if (!projectId) {
			// Set the Project to null,
			// and send the user to the homepage.
			project.value = null;
			goToHomepage();
		} else {
			const id = projectId as string;
			project.value = await ProjectService.get(id);
			if (!project.value) {
				goToHomepage();
			}
		}
	},
	{ immediate: true }
);
```